src/constructors/createGlobalStyle.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+import React from 'react'
+import PropTypes from 'prop-types'
+import hashStr from '../vendor/glamor/hash'
+import StyleSheet, { CONTEXT_KEY } from '../models/StyleSheet'
+import ServerStyleSheet from '../models/ServerStyleSheet'
+import type { Interpolation, Stringifier } from '../types'
+import {
+  CHANNEL,
+  CHANNEL_NEXT,
+  CONTEXT_CHANNEL_SHAPE,
+} from '../models/ThemeProvider'
+
+export default (stringifyRules: Stringifier, css: Function) => {
+  const createGlobalStyle = (
+    strings: Array<string>,
+    ...interpolations: Array<Interpolation>
+  ) =>
+    class extends React.Component {
+      static contextTypes = {
+        [CHANNEL]: PropTypes.func,
+        [CHANNEL_NEXT]: CONTEXT_CHANNEL_SHAPE,
+        [CONTEXT_KEY]: PropTypes.oneOfType([
+          PropTypes.instanceOf(StyleSheet),
+          PropTypes.instanceOf(ServerStyleSheet),
+        ]),
+      }
+      state = { theme: {} }
+      componentId = ''
+      unsubscribeId = -1
+
+      componentWillMount() {
+        const context = this.context[CHANNEL_NEXT]
+        const theme = typeof context !== 'undefined' ? context.getTheme() : {}
+
+        if (typeof context !== 'undefined') {
+          this.setState({ theme })
+
+          this.unsubscribeId = context.subscribe(nextTheme => {
+            this.setState({
+              theme: nextTheme,
+            })
+          })
+        }
+
+        const rules = css(
```

Comment:
this diverges a lot from what we're doing, you'll probably want to use `ComponentStyle`

Reply:
Tried to implement on top of `ComponentStyle`, but found in hard to bend it to my use case (no generated classNames, etc). I went ahead and extracted the required parts to a simplified `GlobalStyle`